@here/harp-geoutils/lib/projection/MercatorProjection.ts

```diff
@@ -92,7 +92,7 @@ class MercatorProjection extends Projection {
         result.y =
             (MercatorProjection.latitudeClampProject(geoPoint.latitudeInRadians) * 0.5 + 0.5) *
             this.unitScale;
-        result.z = geoPoint.altitude ?? 0;
+        result.z = (geoPoint.altitude ?? 0) * this.getScaleFactor(result);
```

Comment:
i don't think this is a good idea. This will enforce applying the scaling factor to Z at projection time, resulting always in different altitudes for objects with large latitude span. So many objects will have oblique planes and the their normal vector will also different.

Reply:
As discussed, I now scale the height on decoding depending on the technique parameter constantHeight, which now is available in all techniques.